freqtrade/exchange/binance.py

```diff
@@ -0,0 +1,26 @@
+""" Binance exchange subclass """
+import logging
+from typing import Dict
+
+from freqtrade.exchange import Exchange
+
+logger = logging.getLogger(__name__)
+
+
+class Binance(Exchange):
+
+    _ft_has: Dict = {
```

Comment:
what prevents you to name it simply "has" ?

Reply:
conflict with  `ccxt.has` - or better - naming it the same makes it un-obvious if I use `exchange.has` - or `exchange._api.has`  - and maybe we could merge ccxt_has and our has in the future to have one "has" (well that sentence turned on me, didn't it?)

Reply:
well `exchange.has`means FT exchange "has" and `exchange._api.has` means ccxt "has". for me it seems eloquent. but nothing against _ft_has honestly so your choice

Reply:
simple 'has' is much more difficult to grep, the results will include many text messages and comments, that should be filtered out additionaly...

Reply:
to put it boldly - we can always rename it later ... it's not user-supplied (at least not by default as a strategy) so it is not set in stone in my eyes.

Reply:
> it's not user-supplied (at least not by default as a strategy)
Let's discuss it more deeper...

As you can see, I also thought of it and partly implemented in that WIP commits for BitMEX subclassing... My I idea was that those exchange 'options' defined in a subclass should be mergeble with options in the config. Similar to strategy options. The reason was the following: a user (or a developer) that tries to use the bot with a new unsupported exchange, could simply define needed options in the config. Of course, they should not be visible explicitely for a user so I thought to merge them deeper, into config['exchange']['internals'], for example... (Thus the name for the options dict in the subclass I used in my experiments.)

However, I thought about this again since that and in the meantime I guess that those options *should not* be merged into config and be available in the config. A such option may appear in several places in the code and it's effect may be too tricky to explain for a config editor, she would still be in need to look at the code in order to understand the effect of the particular option. So, if a user wants to use the bot with a new unsupported exchange, there shall be a developer who adds the new subclass for the exchange with the options for that exchange...

So my current opinion is opposite -- those options should not be exposed to the config. Well, may be _some of them_, like a used window length (either exchange default or max) for responses used in pagination while fetching historical and live ohlcvs...

Reply:
Agreed - as mentioned this is a first step.
Until we have things implemented we can actually expose, the name is not really important.

Ideally we don't need to subclass many exchanges... otherwise we'll need to limit support soonish - since i do not intend to open an account with 120 exchanges...